acceptance/README.md

```diff
@@ -1,164 +1,439 @@
+`bundle update` to install the new gems.
+
+#### VMPooler
+[VMPooler](https://github.com/puppetlabs/vmpooler) is the default hypervisor we
+use. This is only available to Puppet, Inc. employees as it uses internal
+infrastructure. If you have access to a similar setup, then you are welcome to
+use this option with a few values changed. If you are using the Puppet internal
+vmpooler, then you can simply run the acceptance rake tasks. See above about how
+to use the `HOSTS` environment variable to customize the platforms you are
+running tests on.
+
+To use a different pooling api, use
+`--global-config pooling_api=http://customvmpooler/` when you use
+`beaker-hostgenerator` to generate `hosts.yaml`. Make sure you set `HOSTS` to
+the host file you just generated so the automation knows to use that file.
+
+#### Docker
+To test with [the docker hypervisor](https://github.com/puppetlabs/beaker-docker),
+you will want to generate a custom hosts file. You may also need to manually
+edit the file, depending on what kind of tests you are running.
+
+
+The following hosts file uses a master from vmpooler and a docker agent. When
+using rake tasks or beaker to run acceptance, ensure `HOSTS` always points to
+your hosts file.
 
-This corresponds to installing facter and hiera stable from our internal mirror, while installing puppet SHA abcd from a git daemon on my local machine percival.  See below for details on setting up a local git daemon.
-
-Running Tests on the vcloud (now vmpooler)
----------------------------
-
-Only applies to Puppet, Inc. employees. Community members should see the [guide to running the tests on vagrant boxen](#running-tests-on-vagrant-boxen).
-
-This hypervisor has been deprecated: we now use vmpooler. Hosts files have been upgraded as appropriate.
-
-### Authentication
-
-Normally the ci tasks are called from a prepared Jenkins job.
-
-If you are running this on your laptop, you will need this ssh private key in order for beaker to be able to log into the vms created from the hosts file:
-
-https://github.com/puppetlabs/puppetlabs-modules/blob/production/secure/jenkins/id_rsa-acceptance
-https://github.com/puppetlabs/puppetlabs-modules/blob/production/secure/jenkins/id_rsa-acceptance.pub
-
-Please note in acceptance/Rakefile where the ssh key is defaulted to. It may be looking in ~/.ssh/id_rsa-acceptance, but it may want to look in the working directory (e.g. puppet/acceptance).
+```
+bundle exec beaker-hostgenerator --hypervisor docker debian8-64a
+```
+```
+---
+HOSTS:
+  debian8-64-2:
+    docker_cmd:
+      - "/sbin/init"
+    image: debian:8
+    platform: debian-8-amd64
+    packaging_platform: debian-8-amd64
+    docker_image_commands:
+      - cp /bin/true /sbin/agetty
+      - rm -f /usr/sbin/policy-rc.d
+      - apt-get update && apt-get install -y cron locales-all net-tools wget
+    hypervisor: docker
+    roles:
+      - agent
+CONFIG:
+  nfs_server: none
+  consoleport: 443
+```
```

Comment:
I had to add a master node (as puppet's presuite requires it). I don't know if there's a single beaker-hostgenerator command that supports multiple hosts from different hypervisors?